### DISCUSSION TITLE
taro的hook全部变成了undefined

### DISCUSSION THREAD

**Discussion by Peanut429** [NervJS/taro#12098] [Q & A] [answered]
我在项目中添加了react-redux和redux-toolkit之后，项目出现了很多问题，redux可以正常工作，但是taro组件全部报红，显示“View”不能用作 JSX 组件，然后taro中的hook可以导入，但是导入之后全是undefined。

View”不能用作 JSX 组件这样的报错有看到别人说删除@types/hoist-non-react-statics可以解决，删除后确实没有在报这样的错了。

然后我尝试把所有和redux相关的东西全部删除，View”不能用作 JSX 组件这样的报错没有了，但是taro的hook仍然是undefined。
我又尝试新开一个项目，都吧Taro对象打印出来看了一下
![image](https://user-images.githubusercontent.com/29727887/180597462-a3ad0982-b267-403b-a56e-ddc23f0ba365.png)
右侧是新开的一个项目，里面就有hook函数，但是原本的项目里就不存在哪些hook，node_modules删除了重新安装也没用。

package.json中的依赖如下：
      "dependencies": {
		"@babel/runtime": "^7.7.7",
		"@reduxjs/toolkit": "^1.8.3",
		"@tarojs/components": "3.4.14",
		"@tarojs/plugin-framework-react": "3.4.14",
		"@tarojs/react": "3.4.14",
		"@tarojs/runtime": "3.4.14",
		"@tarojs/taro": "3.4.14",
		"classnames": "^2.3.1",
		"dayjs": "^1.11.4",
		"react": "^17.0.0",
		"react-dom": "^17.0.0",
		"react-redux": "^8.0.2",
		"taro-ui": "^3.1.0-beta.2"
	},
	"devDependencies": {
		"@babel/core": "^7.8.0",
		"@tarojs/mini-runner": "3.4.14",
		"@tarojs/plugin-react-devtools": "^3.4.13",
		"@tarojs/webpack-runner": "3.4.14",
		"@types/react": "^17.0.2",
		"@types/webpack-env": "^1.13.6",
		"@typescript-eslint/eslint-plugin": "^5.20.0",
		"@typescript-eslint/parser": "^5.20.0",
		"babel-preset-taro": "3.4.14",
		"eslint": "^8.12.0",
		"eslint-config-taro": "3.4.14",
		"eslint-plugin-import": "^2.12.0",
		"eslint-plugin-react": "^7.8.2",
		"eslint-plugin-react-hooks": "^4.2.0",
		"stylelint": "^14.4.0",
		"typescript": "^4.1.0"
	}

**Comment by Peanut429** [comment, marked as answer]
我之前的taro的版本是3.4.12，更新到3.4.14后删除node_modules在重新安装依赖就正常了

**Comment by AdvancedCat** [comment]
关于 Taro 的问题，移驾到 issues 中反馈吧
